lasagne/tests/layers/test_conv.py

```diff
@@ -566,10 +593,16 @@ def test_with_nones(self, DummyInputLayer, input, kernel, output, kwargs):
                 filter_size=kernel.shape[2:],
                 W=kernel.transpose(1, 0, 2, 3),
                 **kwargs)
-        assert layer.output_shape == (None, output.shape[1], None, None)
+        if 'output_size' not in kwargs or isinstance(kwargs['output_size'],
+                                                     T.Variable):
+            assert layer.output_shape == (None, output.shape[1], None, None)
         actual = layer.get_output_for(input).eval()
         assert actual.shape == output.shape
         assert np.allclose(actual, output)
+        # Check get_output_shape_for for non symbolic output
+        if 'output_size' in kwargs and not isinstance(kwargs['output_size'],
+                                                      T.Variable):
+            assert layer.get_output_shape_for(input.shape) == output.shape
```

Comment:
> Sorry for the wealth of requests. 

No problem at all!

I added your test and squashed. This should be it!
In case it's needed in the future, the alpha version was in bf594106eb977f38b87d253d82960687aeeb7fa2